classes/validation/ValidatorFactory.inc.php

```diff
@@ -0,0 +1,373 @@
+<?php
+/**
+ * @file classes/validation/ValidatorFactory.php
+ *
+ * Copyright (c) 2014-2018 Simon Fraser University
+ * Copyright (c) 2000-2018 John Willinsky
+ * Distributed under the GNU GPL v2. For full terms see the file docs/COPYING.
+ *
+ * @class ValidatorFactory
+ * @ingroup validation
+ *
+ * @brief A factory class for creating a Validator from the Laravel framework.
+ */
+use Illuminate\Container\Container;
+use Illuminate\Filesystem\Filesystem;
+use Illuminate\Translation\FileLoader;
+use Illuminate\Translation\Translator;
+use Illuminate\Validation\DatabasePresenceVerifier;
+use Illuminate\Validation\Factory;
+
+class ValidatorFactory {
+
+	/**
+	 * Create a validator
+	 *
+	 * This is a wrapper function for Laravel's validator factory. It loads the
+	 * necessary dependencies and instantiates Laravel's validation factory, then
+	 * calls the `make` method on that factory.
+	 *
+	 * @param $props array The properties to validate
+	 * @param $rules array The validation rules
+	 * @param $messages array Error messages
+	 * @return Illuminate\Validation\Validator
+	 */
+	static public function make($props, $rules, $messages = []) {
```

Comment:
I'm ambivalent about building all kinds of validators in this function, which feels like it should be a base class or something.

Reply:
`ValidatorFactory` is designed to mimic the recommended approach to [instantiate Laravel's Validation class](https://laravel.com/docs/5.5/validation#manually-creating-validators), since we have to wrap their factory class in [some configuration](https://github.com/NateWr/pkp-lib/blob/i3594_forms/classes/validation/ValidatorFactory.inc.php#L37-L42) and [locale handling](https://github.com/NateWr/pkp-lib/blob/i3594_forms/classes/validation/ValidatorFactory.inc.php#L140) to instantiate a validator.

Instead of having a `ValidatorSize` class, Laravel makes all validation rules available to each validator, and you pass in the values and the rules against which you want to validate them:

```
$validator = Validator::make(
	['title' => 'This is my title!'], // Values
	['title' => 'required|unique:posts|max:255'] // Rules
);
```

I'm open to having the closures which define each rule stored elsewhere. But I was worried that it will get a little bit confusing since our existing `Validator` classes sit alongside `ValidatorFactory`. 


Also, with `use ($validation)` in play here, I thought separating them might make it kind of confusing to know where `$validation` comes from and it would prevent them from being reusable on their own.

Reply:
OK, gotcha.